src/goto-programs/link_to_library.cpp

```diff
@@ -59,30 +116,44 @@ void link_to_library(
         // already added
       }
       else
-        missing_functions.insert(id);
+      {
+        changed = true;
+        added_functions.insert(id);
+
```

Comment:
How about reformulating the control flow structure as
```c
if(has_body || already_added)
 continue;

// add the symbol
changed = true;
added_functions.insert(id);
...
 
```

Reply:
Could do, but then I'd have to touch more code in this PR.